### ISSUE TITLE
[FEDEX] Duty payment set to recipient raise a ACCOUNT.NUMBER.INVALID error

### ISSUE THREAD

**Issue by vangelow** [karrioapi/karrio#697]
**Describe the bug**
When setting the Duty payments to recipient, fedex throws an error that the account number is invalid, when i am not sending an account number at all. Originally in the fedex API, the account number is not required. Also the GUI of fedex allows you to set duty payments to recipient without setting an account number (it's optional)

**To Reproduce**
Send a `POST` request to `/v1/shipments`
```
{
  "carrier_ids": [
    "fedex-test"
  ],
  "customs": {
    "certify": true,
    "commodities": [
      {
        "quantity": 1,
        "title": "Personal documents",
        "value_amount": 1,
        "value_currency": "EUR",
        "weight": 1,
        "weight_unit": "KG"
      }
    ],
    "content_type": "documents",
    "duty": {
      "currency": "EUR",
      "declared_value": 1,
      "paid_by": "recipient"
    },
    "incoterm": "DDU",
    "signer": "John Doe"
  },
  "label_type": "PDF",
  "metadata": {},
  "options": {
    "currency": "EUR",
    "declared_value": 1,
    "preferred_service": "fedex_international_economy"
  },
  "parcels": [
    {
      "dimension_unit": "CM",
      "height": 1,
      "is_document": true,
      "length": 1,
      "weight": 1,
      "weight_unit": "KG",
      "width": 1
    }
  ],
  "payment": {
    "paid_by": "sender"
  },
  "recipient": {
    "address_line1": "Valid addres line 1",
    "city": "Rostock",
    "country_code": "DE",
    "person_name": "Valid person name",
    "postal_code": "Valid postal code",
    "residential": true,
    "validate_location": false
  },
  "service": "fedex_international_economy",
  "shipper": {
    "address_line1": "Valid address line 1",
    "city": "6-October",
    "country_code": "EG",
    "person_name": "John doe 2",
    "postal_code": "Valid postal code",
    "residential": true,
    "validate_location": false
  }
}
```
**Expected behavior**
I expect that the shipment is created and duty payments are set to recipient without invalid account number error.

**Screenshots**
![image](https://github.com/user-attachments/assets/f8cbea02-4490-47e0-8aa1-0651339d1517)

**Additional context**
What makes me think that the bug exists is that i can see in the logs create shipment request sends 2 API calls to fedex,
One at: `https://apis-sandbox.fedex.com/rate/v1/rates/quotes`
And one at: `https://apis-sandbox.fedex.com/ship/v1/shipments`

I can clearly see that the Duty context i sent is only applied to the shipments call and not the rates/quotes call, which makes me think that that is a potential place from where the problem comes from
Examples;

Request for `/shipments` has the `paymentType` set to recipient:
```
      "dutiesPayment": {
        "paymentType": "RECIPIENT",
        "payor": {
          "responsibleParty": {
            "address": {
              "city": "Rostock",
              "countryCode": "DE",
              "postalCode": "Some valid postal code",
              "residential": true,
              "streetLines": [
                "Some valid address line"
              ]
            },
            "contact": {
              "personName": "Some valid person name"
            }
          }
        }
      },
``` 
While the `/rates/quotes` request is sent as:
```
      "dutiesPayment": {
        "paymentType": "SENDER",
        "payor": {
          "responsibleParty": {
            "accountNumber": {
              "value": "THE ACCOUNT NUMBER FROM CARRIER CONNECTION"
            }
          }
        }
```  
Took this from karrio api logs
 - Version [e.g. 2024.6.7]

**Comment by jacobshilitz** [comment]
I think you need to supply the account number of the recipient

    "duty": {
      "currency": "EUR",
      "declared_value": 1,
      "paid_by": "recipient",
      "account_number": "<fedex account number>"
    },
![image](https://github.com/user-attachments/assets/a054e83a-9a63-4474-8dfd-9e187e96100a)

**Comment by vangelow** [comment]
@jacobshilitz Thank you for the suggestion, but even passing the account number i still get an error.
After all with the same settings and data on the fedex website i can do a shipment without an account number when Duty is set to recipient
![image](https://github.com/user-attachments/assets/ae816da8-dedb-47aa-8ca1-3cc052542bbd)

**Comment by ChrisNolan** [comment]
The screenshot you gave Jacob shows the account number as 'nullable' so I'd think it wasn't required.

Interestingly, in the 'shipping_responses.json' schema, whenever the "We are not able to retrieve the message for the warning or error" is shown, it is shown as a 'warning' rather than an error.

The class/type does seem to be setup to have account number be optional as well correctly
```
@s(auto_attribs=True)
class ResponsiblePartyType:
    address: Optional[AddressType] = JStruct[AddressType]
    contact: Optional[ResponsiblePartyContactType] = JStruct[ResponsiblePartyContactType]
    accountNumber: Optional[AccountNumberType] = JStruct[AccountNumberType]
    tins: List[TinType] = JList[TinType]


@s(auto_attribs=True)
class PayorType:
    responsibleParty: Optional[ResponsiblePartyType] = JStruct[ResponsiblePartyType]

```

so... yeah, I'm not sure but there is some extra info at least.

**Comment by vangelow** [comment]
@ChrisNolan I can imagine that in fedex server side the account number is required, but conditionally. I guess the combination of other fields makes it required and I hoped someone has an idea why. If I find something on the topic I will put it here, meanwhile will leave this here in case @danh91 has some idea.

**Comment by danh91** [comment]
Hi @vangelow,
I took a look into this and noticed that the payment account wasn't formatted correctly.
The fix was pushed with patch `2024.9.13`.

Please give it a try and let me know how it goes on your end.
